core/src/Kokkos_Swap.hpp

```diff
@@ -0,0 +1,67 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SWAP_HPP
+#define KOKKOS_SWAP_HPP
```

Comment:
i was looking at something, and realized that here you don't have the following lines: 
```
#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE                                                                            |
#define KOKKOS_IMPL_PUBLIC_INCLUDE                                                                            |
#define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_SWAP                                                            |
#endif
```
and at the bottom, like in other headers, is it intentional?

Reply:
We don't use any private Kokkos headers here so there is no reason to define `KOKKOS_IMPL_PUBLIC_INCLUDE` (and undefining it at the end).